src/index/blockfilterindex.cpp

```diff
@@ -31,6 +31,8 @@ constexpr char DB_FILTER_POS = 'P';
 constexpr unsigned int MAX_FLTR_FILE_SIZE = 0x1000000; // 16 MiB
 /** The pre-allocation chunk size for fltr?????.dat files */
 constexpr unsigned int FLTR_FILE_CHUNK_SIZE = 0x100000; // 1 MiB
+/** Maximum size of the cfheaders cache */
+constexpr size_t CF_HEADERS_CACHE_MAX_SZ{2000};
```

Comment:
Is there any particular reason for limiting the size of the cache? With this numbers it would be full at block 2000000, i.e. in (2000000-630000)/(6 * 24 * 365) ~ 26 years.

Reply:
I wanted to have some limit so that a bug doesn't turn into a memory exhaustion possibility (since we only ever add to this map, and never remove). I don't think the exact number matters. Presumably in 26 years, anyone who wants to serve compact block filters from a v0.21 Bitcoin Core server will be able to patch their own code.

Reply:
I agree that when in doubt, it's better to be overcautious than risking introducing a memory exhaustion bug -- I just wanted to get sure there are no other reasons for the limit that I'm not aware of.